Utilities/DataLoading/EdFi.SmokeTest.Console/Application/SmokeTestConfigurationValidator.cs

```diff
@@ -0,0 +1,107 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+using System;
+using System.IO;
+using System.Text;
+using EdFi.Common.Configuration;
+using EdFi.LoadTools.SmokeTest;
+
+namespace EdFi.SmokeTest.Console.Application
+{
+    public class SmokeTestConfigurationValidator : ISmokeTestConfigurationValidator
+    {
+        private readonly SmokeTestsConfiguration _configuration;
+
+        public SmokeTestConfigurationValidator(SmokeTestsConfiguration configuration)
+        {
+            _configuration = configuration;
+        }
+
+        private bool ValidApiUrl
+        {
+            get => Uri.IsWellFormedUriString(_configuration.ApiUrl, UriKind.Absolute);
+        }
+
+        private bool ValidOAuthUrl
+        {
+            get => Uri.IsWellFormedUriString(_configuration.OAuthUrl, UriKind.Absolute);
+        }
+
+        private bool ValidMetadataUrl
+        {
+            get => _configuration.TestSet == TestSet.NonDestructiveSdk ||
+                   Uri.IsWellFormedUriString(_configuration.MetadataUrl, UriKind.Absolute);
+        }
+
+        private bool ValidSdkLibraryPath
+        {
+            get => _configuration.TestSet == TestSet.NonDestructiveApi || File.Exists(_configuration.SdkLibraryPath);
+        }
+
+        private bool ValidNamespacePrefix
+        {
+            get => _configuration.TestSet != TestSet.NonDestructiveApi ||
+                   Uri.IsWellFormedUriString(_configuration.NamespacePrefix, UriKind.Absolute);
+        }
+
+        public string ErrorText { get; private set; }
+
+        public bool IsValid()
+        {
+            var isValid = ValidApiUrl && ValidOAuthUrl && ValidMetadataUrl && ValidSdkLibraryPath &&
+                          ValidNamespacePrefix;
+
+            if (_configuration.ApiMode == ApiMode.YearSpecific)
+            {
+                isValid = isValid && _configuration.SchoolYear.HasValue;
+            }
+
+            if (!isValid)
+            {
+                ErrorText = CreateErrorText();
+            }
+
+            return isValid;
+        }
+
+        private string CreateErrorText()
+        {
+            var sb = new StringBuilder();
+
+            if (!ValidApiUrl)
+            {
+                sb.AppendLine("'a: apiurl is not a valid URL");
```

Comment:
From a consistency standpoint with the earlier approach, these messages are inserting a space between the short/long command line options. May want to consider picking a style and applying it consistently throughout.

Reply:
this is a legacy artifact